terraform/ec2-examples/distributed-ml-training/main.tf

```diff
@@ -0,0 +1,489 @@
+provider "aws" {
+  region = local.region
+}
+
+data "aws_availability_zones" "available" {}
+data "aws_caller_identity" "current" {}
+
+locals {
+  name   = "ecs-demo-distributed-ml-training"
+  region = "us-east-1"
+
+  vpc_cidr                   = "10.0.0.0/16"
```

Comment:
Removed VPC and service discovery namespace creation. To avoid deploying clusters and ASGs that are not going to be used, I am using resource targeting